netket/experimental/driver/tdvp_common.py

```diff
@@ -103,37 +106,54 @@ def __init__(
         self._dw = None  # type: PyTree
         self._last_qgt = None
         self._integrator = None
-        self._integrator_constructor = None
 
         self._odefun = HashablePartial(odefun_host_callback, self.state, self)
 
         self.error_norm = error_norm
-        self.integrator = integrator
+
+        if integrator is not None:
+            warn_deprecation(
+                "Specifying `integrator` when constructing TDVP is deprecated. Please use `solver` instead."
```

Comment:
```suggestion
                "The keyword argument `integrator` has been renamed to `solver` and "
            "deprecated to improve clarity.\n"
            "Update the keyword argument in your code to avoid breakage in the future."
```

This is a bit more clear